app/models/remote_theme.rb

```diff
@@ -113,6 +113,7 @@ def self.update_theme(
         )
       existing = false
     end
+    theme.has_just_been_created = !existing
```

Comment:
Your solution works, but it is a bit complex. Update of theme is setting `has_just_been_created` which is later used by `SystemThemesManager`.

What would you say if instead, we would hook directly here:

https://github.com/discourse/discourse/blob/main/db/fixtures/600_themes.rb#L5

I am thinking about something like `SystemThemesManager.sync!(has_just_been_created: !theme_exists)`

Reply:
I reworked the fix and removed the need to use an accessor on `Theme` or pass a variable to from the seed file down to `SystemThemesManager.sync!` -- it's now a basic variable in the `SystemThemesManager.sync_theme!` method that checks for the theme existence just before it's imported.